python/cudf/cudf/tests/test_repr.py

```diff
@@ -84,34 +92,23 @@ def test_full_series(nrows, dtype):
     pd.reset_option("display.max_rows")
 
 
+@pytest.mark.parametrize("nrows", [5, 10, 15])
+@pytest.mark.parametrize("ncols", [5, 10, 15])
+@pytest.mark.parametrize("size", [20, 21])
 @pytest.mark.parametrize("dtype", repr_categories)
-@pytest.mark.parametrize("nrows", [0, 1, 2, 9, 20 / 2, 11, 20 - 1, 20, 20 + 1])
-@pytest.mark.parametrize("ncols", [0, 1, 2, 9, 20 / 2, 11, 20 - 1, 20, 20 + 1])
-def test_full_dataframe_20(dtype, nrows, ncols):
-    size = 20
+def test_full_dataframe_20(dtype, size, nrows, ncols):
     pdf = pd.DataFrame(
         {idx: np.random.randint(0, 100, size) for idx in range(size)}
     ).astype(dtype)
     gdf = cudf.from_pandas(pdf)
 
+    pd.options.display.max_rows = int(nrows)
+    pd.options.display.max_columns = int(ncols)
+
     assert pdf.__repr__() == gdf.__repr__()
```

Comment:
Don't call `__repr__()` directly since it's a dunder-method with magic behavior. It's like writing `len(x)` instead of `x.__len__()`. Instead, use:
```suggestion
    assert repr(pdf) == repr(gdf)
```

I've been meaning to make a PR that fixes this globally within cudf and its tests. We do the same thing in our internal definitions of `__repr__` methods that should be calling `repr(x)` on child objects whose representations are a part of the parent object's `repr`, instead of `child_object.__repr__()`.

(This advice does not apply to the fancy IPython representations for HTML / LaTeX. The only way for us to test that without calling the private methods like `_repr_html_` would require IPython's `HTMLFormatter` class, which we should avoid introducing as a pure test dependency.)